src/tools/cephfs/cephfs_shell.py

```diff
@@ -0,0 +1,607 @@
+#!/usr/bin/env python3
+# coding=utf-8
+
+import argparse
+import os
+import sys
+from cmd2 import Cmd, with_argparser
+import cephfs as libcephfs
+import shlex
+import shutil
+import traceback
+import colorama
+import readline
+
+cephfs =  None
+
+def setup_cephfs():
+    """ Mouting a cephfs """
+    global cephfs
+    cephfs = libcephfs.LibCephFS(conffile='/home/admin1/Documents/ceph/build/ceph.conf')
+    cephfs.mount()
+
+def mode_notation(mode):
+    """ """
+    permission_bits = {'0':'---', '1':'--x', '2':'-w-', '3' : '-wx', '4':'r--', '5':'r-x', '6':'rw-', '7':'rwx'}
+    mode = str(oct(mode))
+    notation = '-'
+    if mode[2] == '4':
+        notation = 'd'
+    for i in mode[-3:]:
+        notation += permission_bits[i]
+    return notation
+
+def get_chunks(file_size):
+    chunk_start = 0
+    chunk_size = 0x20000  # 131072 bytes, default max ssl buffer size
+    while chunk_start + chunk_size < file_size:
+        yield(chunk_start, chunk_size)
+        chunk_start += chunk_size
+    final_chunk_size = file_size - chunk_start
+    yield(chunk_start, final_chunk_size)
+
+def to_bytes(string):
+    return bytes(string, encoding="utf-8")
+
+def list_items(dir_name = ''):
+    if not isinstance(dir_name, bytes):
+        file_name = to_bytes(dir_name)
+    if dir_name == '':
+        d = cephfs.opendir(cephfs.getcwd())
+    else :
+        d = cephfs.opendir(dir_name)
+    dent = cephfs.readdir(d)
+    items = []
+    while dent:
+        if dent.d_name not in [b".", b".."]:
+            items.append(dent)
+        dent = cephfs.readdir(d)
+    cephfs.closedir(d)
+    return items
+
+def print_details(file_name, flag):
+    if not isinstance(file_name, bytes):
+      file_name = to_bytes(file_name)
+    info = cephfs.stat(file_name)
+    print(mode_notation(info.st_mode), info.st_size, info.st_uid, info.st_gid, info.st_mtime, colorama.Fore.BLUE * flag + file_name.decode("utf-8").rsplit('/',1)[1] + '/'*flag + colorama.Style.RESET_ALL * flag,sep='\t')
+
+def word_len(word):
+    """Returns the word lenght, minus any color codes."""
+    if word[0] == '\x1b':
+        return len(word) - 9  
+    return len(word)
+
+def print_list(words, termwidth=79):
+    if not words :
+        return
+    width = max([word_len(word) for word in words])
+    nwords = len(words)
+    ncols = max(1, (termwidth + 1) // (width + 1))
+    nrows = (nwords + ncols - 1) // ncols
+    for row in range(nrows):
+        for i in range(row, nwords, nrows):
+            word = words[i]
+            if word[0] == '\x1b':
+                print('%-*s' % (width + 9, words[i]),
+                           end='\n' if i + nrows >= nwords else ' ')
```

Comment:
Line break is incorrect. Don't know how Python code is handled in this subproject, but in the Ceph Dashboard we have a linter that makes sure the Python code is structured correct.